### ISSUE TITLE
Does not activate on Debian 12 (43.6)

### ISSUE THREAD

**Issue by pabl0** [nick-shmyrev/improved-osk-gnome-ext#62]
After installing and enabling the extension (and restarting the GNOME session), I still get the standard (and sadly, almost useless) stock GNOME OSK. If I toggle "Show statusbar icon" in the settings from the extension manager, it will indeed toggle the icon. However, touching the icon will just activate the same old basic OSK.

It is not clear from documentation, but I assume the newly installed extension should replace the default OSK without any further actions from the user?

The only relevant log entries I can find in the system journal:
``
JS ERROR: TypeError: Main.keyboard._keyboard is null
toggleOSK@/home/pablo/.local/share/gnome-shell/extensions/improvedosk@nick-shmyrev.dev/extension.js:67:7
_init/<@/home/pablo/.local/share/gnome-shell/extensions/improvedosk@nick-shmyrev.dev/extension.js:60:9
``

The extension was installed from the command line with 

``
gnome-extension install improvedosknick-shmyrev.dev.v18.shell-extension.zip
``
BTW, I originally installed the original implementation from Sebastian Luebke, but uninstalled it immediately after realizing my mistake, and I don't believe any files from that remains on my system.

**Environment:**
 - Distro: Debian 12.1
 - Gnome version: 43.6
 - Windowing system: Wayland

**Comment by pabl0** [comment]
Uh, sorry, somehow I missed that this seems to be a duplicate of #61 for Fedora 38.

**Comment by pabl0** [comment]
Of course, should have mentioned that I use the Finnish keyboard layout (fi), which apparently is still missing the extended keys. If I switch to English (US), I get all the new goodness. so I guess it is just a matter of adding the new keys to the layout. Are those non-alphanumeric keys really that different between any countries using the same physical ISO layout, would the same change as for Hungarian work?

**Comment by jimjamz** [comment]
I'm on a similar environment (Debian 12, Gnome 43.9, Wayland) and I have only the English (US) keyboard installed, and I _still_ don't see the improved OSK layout, only the default OSK.

I've tried multiple installation methods, including:
- via [the GNOME extensions page](https://extensions.gnome.org/extension/4413/improved-osk/)
- cloning the repo and running the package installation script
- installing as a global extension

Is it _just_ the keyboard layout that has to be set to English (US) or does it require also the Regional Language and Formats to also be English (US)? Currently, I have the latter two set to English (UK), but should this matter?  These values cannot actually be changed within the settings UI of Debian.

Due to the issues in the package and installation bash scripts, it's not actually possible to run the scripts successfully without fixing them first, such as:
`/etc/dconf/db/local.d: 00-extensions: Key file does not start with a group`

As in many other open issues, after enabling the extension and clicking on the icon in the status bar, `journalctl /usr/bin/gnome-shell -f -o cat` reports:

```
JS ERROR: TypeError: Main.keyboard._keyboard is null
toggleOSK@/home/jimjamz/.local/share/gnome-shell/extensions/improvedosk@nick-shmyrev.dev/extension.js:67:7
_init/<@/home/jimjamz/.local/share/gnome-shell/extensions/improvedosk@nick-shmyrev.dev/extension.js:50:11
```
